components/style/selector_matching.rs

```diff
@@ -179,11 +171,14 @@ impl Stylist {
             ($style_rule: ident, $priority: ident) => {
                 if $style_rule.declarations.$priority.len() > 0 {
                     for selector in &$style_rule.selectors {
-                        let map = match selector.pseudo_element {
-                            None => &mut element_map,
-                            Some(PseudoElement::Before) => &mut before_map,
-                            Some(PseudoElement::After) => &mut after_map,
+                        let map = if let Some(ref pseudo) = selector.pseudo_element {
+                            self.pseudos_map.entry(pseudo.clone())
+                                            .or_insert(PerPseudoElementSelectorMap::new())
```

Comment:
Is `PerPseudoElementSelectorMap::new()` expensive? Maybe use `.or_insert_with(PerPseudoElementSelectorMap::new)` instead.